server/neptune/workflows/activities/conftest.go

```diff
@@ -1,33 +1,129 @@
 package activities
 
 import (
+	"bytes"
 	"context"
+	"fmt"
 	"github.com/hashicorp/go-version"
 	"github.com/runatlantis/atlantis/server/core/config/valid"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/command"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/temporal"
+	"go.temporal.io/sdk/activity"
+	"strings"
 )
 
+type asyncClient interface {
+	RunCommand(ctx context.Context, request *command.RunCommandRequest, options ...command.RunOptions) error
+}
+
+type fileValidator interface {
+	Stat(name string) error
+}
+
+var NoColorFlag = command.Flag{
+	Value: "-no-color",
+}
+
 type conftestActivity struct {
 	DefaultConftestVersion *version.Version
-	ConftestClient         *command.AsyncClient
+	ConftestClient         asyncClient
 	StreamHandler          streamer
 	Policies               valid.PolicySets
+	FileValidator          fileValidator
 }
 
 type ConftestRequest struct {
 	Args        []command.Argument
 	DynamicEnvs []EnvVar
 	JobID       string
-	Version     string
 	Path        string
 	ShowFile    string
 }
 
 type ConftestResponse struct {
-	Output string
+	FailedPolicies map[string]valid.PolicySet
 }
 
 func (c *conftestActivity) Conftest(ctx context.Context, request ConftestRequest) (ConftestResponse, error) {
-	// TODO: Implement conftest activity
-	return ConftestResponse{}, nil
+	cancel := temporal.StartHeartbeat(ctx, temporal.HeartbeatTimeout)
+	defer cancel()
+
+	// validate terraform show file exists
+	showFile := request.ShowFile
+	if err := c.FileValidator.Stat(showFile); err != nil {
+		return ConftestResponse{}, err
+	}
+
+	envs, err := getEnvs(request.DynamicEnvs)
+	if err != nil {
+		return ConftestResponse{}, err
+	}
+
+	var policyNames []string
+	var totalCmdOutput []string
+	failedPolicies := make(map[string]valid.PolicySet)
+
+	// run each policy separately to track which pass and fail
+	for _, policy := range c.Policies.PolicySets {
```

Comment:
Why not separate activities?

Reply:
I was debating it too, but I figured it would be better to simplify the activity by avoiding passing through the log streaming channel within each policy run to stream all the outputs. Also from an admin UX perspective, when debugging, I thought it would be better to just look for one total conftest activity rather than attempt to read through some N number of them. In general, it made more sense to me to consolidate all policies to one activity. Let me know what you think.

Reply:
ok thats fine